docs/ELIGIBILITY_FLOW.md

```diff
@@ -0,0 +1,134 @@
+# Eligibility Flow
+
+The eligibility flow is the sequence of events used to determine if an upkeep is
+considered eligible to perform. This involves some input on a regular interval,
+a series of pre-processing, checking via RPC calls, collecting data from HTTP
+calls, and finally some type of post-processing. The output is expected to be
+only upkeeps eligible to be performed on chain.
+
+Required data includes a collection of active, configured upkeeps with an
+unknown eligibility status.
+
+[Generic Sequence Diagram](./diagrams/generic_ticker_sequence.md)
+
+## Basic Value Types
+
+To maintain chain agnostic behavior, the following data types should be
+abstracted from the plugin as much as possible. That is, the less detail the
+plugin needs from each of the following value types, the better.
+
+### ConfiguredUpkeep
+
+A configured upkeep represents all information that identifies a single upkeep
+as registered by the user including upkeep id, trigger configuration, 
+check data, etc.
+
+```golang
+type UpkeepIdentifier []byte
+
+type ConfiguredUpkeep struct {
+    // ID uniquely identifies the upkeep
+    ID UpkeepIdentifier
+    // Type is the event type required to initiate the upkeep
+    Type int
```

Comment:
We'll be able to derive the type from the ID, could be redundant as additional field

Reply:
The idea in this struct is to pack the type and id together in the registry, but only split it apart once in the go code and pass them around as two values.